### PR TITLE
Run updated eslint preset

### PR DESCRIPTION
**Pull request by kaisermann**
#### What problem is this solving?

- Update tooling to latest version
- Run eslint/prettier on the project

Disclaimer: 

- Switched some `enum`s to `const enum`s.

### CODE REVIEW

File: react/components/SKUSelector/Wrapper.tsx [hunk 1]
```diff
@@ -69,15 +76,21 @@ const getVariationsFromSpecifications = (
...
-  shouldNotShow: boolean,
+const useVariations = ({
+  skuItems,
+  skuSpecifications,
+  shouldNotShow,
+  visibleVariations,
+}: {
+  skuItems: ProductItem[]
+  skuSpecifications: SkuSpecification[]
+  shouldNotShow: boolean
```
Comment (Klynger) [inline react/components/SKUSelector/Wrapper.tsx:91]: Just to confirm, these props can't be `undefined`, right?

File: react/components/SKUSelector/components/SKUSelector.tsx [hunk 2]
```diff
@@ -71,21 +73,29 @@ const isSkuAvailable = compose<
...
-  variationCount: number,
+const showItemAsAvailable = ({
+  possibleItems,
+  selectedVariations,
+  variationCount,
+  isSelected,
+}: {
+  possibleItems: SelectorProductItem[]
+  selectedVariations: SelectedVariations
+  variationCount: number
```
Comment (Klynger) [inline react/components/SKUSelector/components/SKUSelector.tsx:84]: Same as the other comment

File: react/typings/storefront.d.ts [hunk 3]
```diff
@@ -1,13 +1,16 @@
 import { FunctionComponent } from 'react'
 
+type GenericObject = Record<string, any>
```
Comment (Klynger) [inline react/typings/storefront.d.ts:3]: ✨

File: react/components/SKUSelector/Wrapper.tsx (hunk 1 above)
Comment (kaisermann) [reply react/components/SKUSelector/Wrapper.tsx:91]: Don't know, but even if they are `undefined` they will always be passed as `undefined` (lol)

**Pull request merged**